### DISCUSSION TITLE
[OP Stack Deployment] Configuring the sequencer on custom L2 to cover L1 fees for users

### DISCUSSION THREAD

**Discussion by surhud004** [ethereum-optimism/developers#637] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hello OP Stack team,

I’m running a custom L2 chain locally connected to the Holesky testnet on L1. In this setup, I’ve configured the L2 chain to set all L2 transaction fees to zero.

### Issue
However, even though L2 transaction fees are zero, the sender account is still being charged the L1 fee component for transactions. As a result, users still need to have at least some gas tokens in their accounts to initiate transactions. This requirement complicates the onboarding experience for users, as they need to acquire gas tokens from a faucet or enable it in their wallet. Ideally, users should be able to connect to L2 and execute transactions immediately without any extra setup, including managing the gas costs.

### Expected Behavior
Users connect to the L2 network and can directly perform transactions without needing any gas tokens or ETH or additional steps. The sequencer should ideally cover the L1 fee component on behalf of the user.

### Request for Assistance
I believe there may be a way to configure the sequencer to absorb these L1 fees on the user’s behalf. Could you guide me through the relevant parts of the codebase where I can adjust this behavior? Any advice on where this configuration might reside and changes, I’d need to make would be greatly appreciated.

Thank you for your help and for all the work on OP Stack!

### Logs

```
cumulativeGasUsed: 96040,
effectiveGasPrice: 0,
from: '0x697ae170f1cbbb848612f9d8174b29fcca2ef0b4',
gasUsed: 52222,
l1BaseFeeScalar: '0xf4240',
l1BlobBaseFee: '0x1',
l1BlobBaseFeeScalar: '0x0',
l1Fee: '0x2bc0',       ====> this L1 fee component - I would like the sequencer to handle instead of the sender account
l1GasPrice: '0x7',
l1GasUsed: '0x640',
```
<br/>

#### Sending 50 custom ERC-20 tokens with zero gas token as balance of sender - failed transaction
<br/>

![image](https://github.com/user-attachments/assets/221b3812-24db-49db-af20-0f7def95d9c4)
![image](https://github.com/user-attachments/assets/366ff067-4ed9-40ba-a4e4-bcc76e8313cd)

**Comment by Skuycrot** [comment]
Goo

**Comment by Chomtana** [comment]
Can you try settings both `_overhead` and `_scalar` to 0 in the `setGasConfig` function of `SystemConfigProxy` contract

![image](https://github.com/user-attachments/assets/6db5c408-c270-43e2-a830-c81ff334117a)

**Reply by surhud004** [reply]
Hey @Chomtana thank you for your quick response. I already configured the above two values to zero, however, it still does not resolve the issue. Could you please guide me through the codebase to locate the sequencer gas fees calculation logic? thanks!

**Comment by surhud004** [comment]
Hey @Chomtana thank you for your quick response. I already configured the above two values to zero, however, it still does not resolve the issue. Could you please guide me through the codebase to locate the sequencer gas fees calculation logic? thanks!

**Comment by surhud004** [comment]
Hi @tynes could you please take a look at this issue I am still running into? I read your document about the L1 Fee Abstraction, however, is there a way to configure the L1 Fee in the native code such that the sequencer handles the fees on behalf of the sender account on L2? Could you please guide me through the codebase as to where I can find this config to modify? thanks!
